src/event/server_events.c

```diff
@@ -300,7 +300,8 @@ sv_ev_room_history(ProfMessage* message)
     ProfMucWin* mucwin = wins_get_muc(message->from_jid->barejid);
     if (mucwin) {
         // if this is the first successful connection
-        if (ev_is_first_connect()) {
+        // or for some reason the `last_msg_timestamp` is not initialized
+        if (ev_is_first_connect() || !mucwin->last_msg_timestamp) {
```

Comment:
This seems not right to me.
We want to set it inside this if loop actually?

Reply:
d53c477e4 changed this btw.

Reply:
> [d53c477](https://github.com/profanity-im/profanity/commit/d53c477e4e2589e15267fe2b3de4bc59fe32aa3e) changed this btw.

maybe it did, but now we have cases where `g_date_time_compare()` is called with first argument set to `NULL`, which means that something is off.

> This seems not right to me.
> We want to set it inside this if statement actually?

What is not right here? We want to initialize `mucwin->last_msg_timestamp` if it is `NULL`.

Reply:
Somehow I'm too confused right now to completely understand it :) But I'll just trust you on this ;)